pkg/transaction/eventgenerator.go

```diff
@@ -191,6 +200,123 @@ func (e *Generator) CreateEvents(summaryEvent LogEvent, detailEvents []LogEvent,
+	}
+
+	// get proxy information
+	if summaryEvent.TransactionSummary.Proxy != nil {
+		apiID = summaryEvent.TransactionSummary.Proxy.ID
+		stage = summaryEvent.TransactionSummary.Proxy.Stage
+		e.logger.
+			WithField("apiID", apiID).
+			WithField("stage", stage).
+			Debug("transaction summary proxy information")
+	} else {
+		return nil
+	}
+
+	// get the managed application
+	managedApp := cacheManager.GetManagedApplicationByName(appName)
+	if managedApp == nil {
+		e.logger.
+			WithField("appName", appName).
+			Debug("could not get managed application by name")
+		return nil
+	}
+	e.logger.
+		WithField("appName", appName).
+		WithField("managed app name", managedApp.Name).
+		Debug("managed application info")
+
+	// get the access request
+	accessRequest := transutil.GetAccessRequest(cacheManager, managedApp, apiID, stage)
+	if accessRequest == nil {
+		e.logger.
+			Debug("could not get access request")
+		return nil
+	}
+	e.logger.
+		WithField("managed app name", managedApp.Name).
+		WithField("apiID", apiID).
+		WithField("stage", stage).
+		WithField("access request name", accessRequest.Name).
+		Debug("managed application info")
+
+	// get subscription info
+	subscription := &Subscription{
+		ID:   unknown,
+		Name: unknown,
+	}
+
+	subscriptionObj := transutil.GetSubscription(cacheManager, accessRequest)
+	if subscriptionObj == nil {
+		e.logger.Debug("could not get subscription")
+		return nil
+	}
+
+	subscription.ID = transutil.GetSubscriptionID(subscriptionObj)
+	subscription.Name = subscriptionObj.Name
+	e.logger.
+		WithField("subscription ID", subscription.ID).
+		WithField("subscription name", subscription.Name).
+		Debug("subscription information")
+
+	// get application info
+	appID := unknown
+	application := &Application{
+		ID:   appID,
+		Name: appName,
+	}
+
+	consumerOrgID := unknown
+
+	if managedApp != nil && subscriptionObj != nil {
+		appID, appName = transutil.GetConsumerApplication(managedApp)
+		application.ID = appID
+		application.Name = appName
+		e.logger.
+			WithField("application ID", application.ID).
+			WithField("application name", application.Name).
+			Debug("application information")
+
+			// try to get consumer org ID from the managed app first
+		consumerOrgID = transutil.GetConsumerOrgID(managedApp)
+		if consumerOrgID == "" {
+			e.logger.Debug("could not get consumer org ID from the managed app, try getting consumer org ID from subscription")
+			// if we can't get it from the managed app, try to get the consumer org ID from the subscription
+			consumerOrgID = transutil.GetConsumerOrgIDFromSubscription(subscriptionObj)
+			if consumerOrgID == "" {
+				e.logger.Debug("could not get consumer org ID from the subscription")
+				return nil
+			}
+		}
+		e.logger.
+			WithField("consumer org ID", consumerOrgID).
+			Debug("consumer org ID ")
+	}
+
+	// Update consumer details with Org, Application and Subscription
+	consumerDetails := &ConsumerDetails{
```

Comment:
just return the object here rather than creating a var?

Reply:
👍🏾